src/frontend-samples/issues-sample/IssuesApi.ts

```diff
@@ -36,37 +35,25 @@ export default class IssuesApi {
     return labels.map((label, index) => ({ ...label, id: instanceKeys[index].id }));
   }
 
-  public static decoratorIsSetup() {
-    return (null != this._issuesPinDecorator);
-  }
-
   public static setupDecorator() {
-    if (undefined === this._issuesPinDecorator)
-      this._issuesPinDecorator = new MarkerPinDecorator();
+    return new MarkerPinDecorator();
   }
 
-  public static addDecoratorPoint(issue: IssueGet, pinImage: HTMLImageElement, title?: string, description?: string, onMouseButtonCallback?: any) {
+  public static addDecoratorPoint(decorator: MarkerPinDecorator, issue: IssueGet, pinImage: HTMLImageElement, title?: string, description?: string, onMouseButtonCallback?: any) {
```

Comment:
the decorator param can be removed because it can just rely on the static decorator. The same idea will be propogated through the rest of the methods.